cpp.mk

```diff
@@ -136,7 +136,7 @@ $${BUILD_DIR}/$1/coverage.xml : $${BUILD_DIR}/$1/.$1_already_run
 .PHONY : coverage_check_$1
 coverage_check_$1 : $${BUILD_DIR}/$1/coverage.xml
 	@mkdir -p $${BUILD_DIR}/scratch/
-	@xmllint --xpath '//class[@line-rate!="1.0"]/@filename' $$< \
+	@xmllint --xpath '//*[count(lines/line) > 0 and @line-rate != "1.0"]/@filename' $$< \
```

Comment:
Is there any need to change from `class` to `*`?  I'd probably leave as `class` if that's still acceptable (as it's more robust).